pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/grpc/PipelinesTest.java

```diff
@@ -0,0 +1,585 @@
+            final var builder = Pipelines.<String, String>bidiStreaming()
+                    .method(sink -> client)
+                    .mapResponse(Bytes::wrap)
+                    .respondTo(replies);
+
+            assertThatThrownBy(builder::build).isInstanceOf(IllegalStateException.class);
+        }
+
+        @Test
+        void methodIsRequired() {
+            final var builder = Pipelines.<String, String>bidiStreaming()
+                    .mapRequest(Bytes::asUtf8String)
+                    .mapResponse(Bytes::wrap)
+                    .respondTo(replies);
+
+            assertThatThrownBy(builder::build).isInstanceOf(IllegalStateException.class);
+        }
+
+        @Test
+        void responseMapperIsRequired() {
+            final var builder = Pipelines.<String, String>bidiStreaming()
+                    .mapRequest(Bytes::asUtf8String)
+                    .method(sink -> client)
+                    .respondTo(replies);
+
+            assertThatThrownBy(builder::build).isInstanceOf(IllegalStateException.class);
+        }
+
+        @Test
+        void respondToIsRequired() {
+            final var builder = Pipelines.<String, String>bidiStreaming()
+                    .mapRequest(Bytes::asUtf8String)
+                    .method(sink -> client)
+                    .mapResponse(Bytes::wrap);
+
+            assertThatThrownBy(builder::build).isInstanceOf(IllegalStateException.class);
+        }
+
+        @Test
+        void nullSubscriptionThrowsNPE() {
+            final var pipeline = Pipelines.<String, String>bidiStreaming()
+                    .mapRequest(Bytes::asUtf8String)
+                    .method(sink -> client)
+                    .mapResponse(Bytes::wrap)
+                    .respondTo(replies)
+                    .build();
+
+            assertThatThrownBy(() -> pipeline.onSubscribe(null))
+                    .isInstanceOf(NullPointerException.class);
+        }
+
+        @Test
+        void onCompleteNextThrowsISE() {
+            final var pipeline = Pipelines.<String, String>bidiStreaming()
+                    .mapRequest(Bytes::asUtf8String)
+                    .method(sink -> {
+                        lenient().doAnswer(invocation -> {
+                            final var msg = invocation.getArgument(0, String.class);
+                            sink.onNext(msg.toUpperCase());
+                            return null;
+                        }).when(client).onNext(any());
+                        return client;
+                    })
+                    .mapResponse(Bytes::wrap)
+                    .respondTo(replies)
+                    .build();
+
+            pipeline.onSubscribe(mock(Flow.Subscription.class));
+            pipeline.onNext(Bytes.wrap("hello"));
+            pipeline.onNext(Bytes.wrap("world"));
+            pipeline.onComplete();
+            pipeline.onNext(Bytes.wrap("!!!"));
+            verify(replies).onError(any(IllegalStateException.class));
```

Comment:
While I realize that the purpose of the test is to verify the behavior after the `onComplete` is invoked, I'm wondering whatever happens on the `replies` side before or immediately after the `onComplete` call? It might be worth `verify()`'ing that too. Otherwise, the behavior remains undefined because the pipeline implementation might invoke `replies.onNext()`, or it might not, and nobody would know for sure if it does (and whether it should or shouldn't do that) just before the third `onNext()` call at line 255.)